graphs/graph_modules/utils.py

```
@@ -0,0 +1,307 @@
+
+def is_valid_date_string(date_string: str) -> bool:
+    """
+    Check if a string is a valid date in the format YYYY-MM-DD.
+
+    :param date_string: The string to check
+    :return: True if the string is a valid date, False otherwise
+    """
+    try:
+        if not isinstance(date_string, str):
+            return False
+        if not date_string.count('-') == 2:
```

Ruff

Use `date_string.count("-") != 2` instead of `not date_string.count("-") == 2`

Replace with `!=` operator

(SIM201)